packages/edit-site/src/components/sync-state-with-url/use-init-edited-entity-from-url.js

```diff
@@ -248,9 +249,14 @@ export default function useInitEditedEntityFromURL() {
 		useResolveEditedEntityAndContext( params );
 
 	const { setEditedEntity } = useDispatch( editSiteStore );
+	const { __unstableSetEditorMode, resetZoomLevel } = unlock(
+		useDispatch( blockEditorStore )
+	);
 
 	useEffect( () => {
 		if ( isReady ) {
+			__unstableSetEditorMode( 'edit' );
+			resetZoomLevel();
```

Comment:
Is this still needed? It feels like a very weird place for this action to be called (sync hook)

Reply:
I don't think the requirement to reset the zoom level when you move to a new entity has changed.

I struggled to find a suitable location for this. I assume the concern is that we're calling a UI-specific hook (i.e. zoom level) in a hook which is primarily concerned with initialisation of editor state?

Any more information on your concerns may help to refactor this.

Reply:
I actually removed this in a PR, so I could have introduced some regressions possibly. Based on your description, I would say we have two options:

- Potentially add this call to the before calling onNavigateEntity or something. (maybe a bit fragile)
- Potentially add this to `EditorProvider` when the "post" prop (id, type) change.